msbuild/Xamarin.Mac.Tasks.Core/Tasks/ArchiveTaskBase.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System;
+using System.IO;
+using System.Diagnostics;
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+using Xamarin.MacDev;
+using Xamarin.MacDev.Tasks;
+
+namespace Xamarin.Mac.Tasks.Tasks
+{
+    public abstract class ArchiveTaskBase : Xamarin.MacDev.Tasks.ArchiveTaskBase
+    {
+        public override bool Execute ()
+        {
+            var archiveDir = CreateArchiveDirectory ();
+
+            try {
+                var plist = PDictionary.FromFile (Path.Combine (AppBundleDir.ItemSpec, "Contents", "Info.plist"));
+                var productsDir = Path.Combine (archiveDir, "Products");
+
+                // TODO: Archive the OnDemandResources.
+                // I don't have any on hand, so I can't use an Xcode-created archive as reference.
```

Comment:
No need for OnDemandResources for macOS. If macOS apps *do* support this, we don't currently support it in the IDE's archiving logic, so don't worry about it.